internal/interfaces/http/tdex_connect_handler.go

```diff
@@ -46,52 +54,97 @@ func NewTdexConnectService(
 		serverPort:        serverPort,
 		macaroonPath:      macaroonPath,
 		certPath:          certPath,
+		repoManager:       repoManager,
 	}
 }
 
 func (t *tdexConnect) TdexConnectHandler(w http.ResponseWriter, req *http.Request) {
-	styleFile, err := ioutil.ReadFile(tdexConnectTemplateCssFile)
-	if err != nil {
-		log.Errorln(err.Error())
-
-		http.Error(w, http.StatusText(500), 500)
-		return
-	}
-
-	macaroon := t.macaroonBytes
-	if t.macaroonBytes == nil {
-		macaroon = readFile(t.macaroonPath)
-		t.macaroonBytes = macaroon
-	}
-
+	ctx := context.Background()
+	walletStatus := t.walletUnlockerSvc.IsReady(ctx)
 	cert := t.certBytes
 	if t.certBytes == nil {
 		cert = readFile(t.certPath)
 		t.certBytes = cert
 	}
 
 	connectUrl, err := tdexdconnect.EncodeToString(
-		t.serverPort, cert, macaroon,
+		t.serverPort, cert, nil,
 	)
 	if err != nil {
 		log.Errorln(err.Error())
 		http.Error(w, http.StatusText(500), 500)
 	}
 
-	data := Page{
-		Title: tdexConnectHtmlTitle,
-		Style: template.CSS(styleFile),
-		Url:   template.URL(connectUrl),
-	}
-
-	//walletStatus := t.walletUnlockerSvc.IsReady(context.Background())
-	//if walletStatus.Initialized {
-	//	TODO check password
-	//}
+	method := req.Method
+	if method == http.MethodGet {
+		styleFile, err := ioutil.ReadFile(tdexConnectTemplateCssFile)
+		if err != nil {
+			log.Errorln(err.Error())
+			http.Error(w, http.StatusText(500), 500)
+
+			return
+		}
+
+		jsFile, err := ioutil.ReadFile(tdexConnectTemplateJsFile)
+		if err != nil {
+			log.Errorln(err.Error())
+			http.Error(w, http.StatusText(500), 500)
+
+			return
+		}
+
+		data := Page{
+			Title:               tdexConnectHtmlTitle,
+			Style:               template.CSS(styleFile),
+			Js:                  template.JS(jsFile),
+			Url:                 template.URL(connectUrl),
+			IsWalletInitialised: walletStatus.Initialized,
+		}
+
+		if err = template.Must(template.ParseFiles(tdexConnectTemplateFile)).Execute(w, data); err != nil {
+			log.Errorln(err.Error())
+			http.Error(w, http.StatusText(500), 500)
+		}
 
-	if err = template.Must(template.ParseFiles(tdexConnectTemplateFile)).Execute(w, data); err != nil {
-		log.Errorln(err.Error())
-		http.Error(w, http.StatusText(500), 500)
+		return
+	} else {
+		if walletStatus.Initialized {
+			password := req.Header.Get("password")
+
+			vault, err := t.repoManager.VaultRepository().GetOrCreateVault(ctx, nil, "", nil)
+			if err != nil {
+				log.Errorln(err.Error())
+				http.Error(w, http.StatusText(500), 500)
+
+				return
+			}
+
+			if _, err := wallet.Decrypt(wallet.DecryptOpts{
+				CypherText: vault.EncryptedMnemonic,
+				Passphrase: password,
+			}); err != nil {
+				log.Debugf(err.Error())
+				http.Error(w, "wrong password", 500)
+
+				return
+			}
+
+			macaroon := t.macaroonBytes
+			if t.macaroonBytes == nil {
+				macaroon = readFile(t.macaroonPath)
```

Comment:
what happens if the daemon is started with NO_MACAROONS ? will be empty?

Reply:
If it is POST, it means we are passing password and then macaroon will be returner, otherwise cert will be returned.